src/status_im/contexts/wallet/account/tabs/view.cljs

```diff
@@ -7,18 +7,24 @@
     [status-im.contexts.wallet.common.activity-tab.view :as activity]
     [status-im.contexts.wallet.common.collectibles-tab.view :as collectibles]
     [status-im.contexts.wallet.common.empty-tab.view :as empty-tab]
-    [utils.i18n :as i18n]))
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
 
 (defn view
   [{:keys [selected-tab]}]
-  [rn/view {:style {:flex 1}}
-   (case selected-tab
-     :assets       [assets/view]
-     :collectibles [collectibles/view]
-     :activity     [activity/view]
-     :permissions  [empty-tab/view
-                    {:title        (i18n/label :t/no-permissions)
-                     :description  (i18n/label :t/no-collectibles-description)
-                     :placeholder? true}]
-     :dapps        [dapps/view]
-     [about/view])])
+  (let [collectible-list (rf/sub [:wallet/current-viewing-account-collectibles])]
+    [rn/view {:style {:flex 1}}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view
+                      {:collectibles         collectible-list
+                       :on-collectible-press (fn [id]
+                                               (rf/dispatch [:wallet/get-collectible-details id])
+                                               (rf/dispatch [:navigate-to :wallet-collectible]))}]
```

Comment:
I had to return to the approach of passing props to `collectibles/view` instead of just subscribing inside.

The reason is this is a general component, and make it able to handle the following by itself:
- All collectibles display (in wallet home)
- Collectibles only for the current account (when an account is picked)
- Collectibles being filtered (in the send asset screen)
Was hard, so I prefered to parameterize it.

Not as clean as before, but not too bad, and it's more reusable now.

Reply:
you could also just make the generic component and then make the various connected components that have the subs inside and handle the data connection etc. Looks good either way 👍

Reply:
Yeah, I think currently is ok, maybe for a later refactoring, we can consider it :)